src/Compilers/CSharp/Portable/Symbols/Source/SourceNamespaceSymbol.cs

```diff
@@ -556,6 +540,216 @@ public void Add(NamespaceOrTypeSymbol symbol)
+                    }
+                    if (!_dictionary.ContainsKey(microsoft))
+                    {
+                        Add(makeEmptyNamespace(microsoft));
+                    }
+                    return;
+                }
+
+                switch (containingNamespace.Name)
+                {
+                    case system:
+                        if (containingNamespace.ContainingNamespace.IsGlobalNamespace && !_dictionary.ContainsKey(runtime))
+                        {
+                            Add(makeEmptyNamespace(runtime));
+                        }
+                        return;
+                    case runtime:
+                        if (isSystem(containingNamespace.ContainingNamespace) && !_dictionary.ContainsKey(compilerServices))
+                        {
+                            Add(makeEmptyNamespace(compilerServices));
+                        }
+                        return;
+                    case microsoft:
+                        if (containingNamespace.ContainingNamespace.IsGlobalNamespace && !_dictionary.ContainsKey(codeAnalysis))
+                        {
+                            Add(makeEmptyNamespace(codeAnalysis));
+                        }
+                        return;
+                    case compilerServices:
+                        if (isRuntime(containingNamespace.ContainingNamespace) && !_dictionary.ContainsKey("NonNullTypesAttribute"))
+                        {
+                            Add(InjectedNonNullTypesAttributeSymbol.Create(containingNamespace));
+                        }
+                        return;
+                    case codeAnalysis:
+                        if (isMicrosoft(containingNamespace.ContainingNamespace) && !_dictionary.ContainsKey("EmbeddedAttribute"))
+                        {
+                            Add(InjectedEmbeddedAttributeSymbol.Create(containingNamespace));
+                        }
+                        return;
+                }
+
+                InjectedNamespaceSymbol makeEmptyNamespace(string name)
+                {
+                    return new InjectedNamespaceSymbol(containingNamespace.ContainingModule, containingNamespace, name);
+                }
+
+                bool isSystem(NamespaceSymbol symbol)
+                {
+                    return symbol.Name == system && symbol.ContainingNamespace.IsGlobalNamespace;
+                }
+
+                bool isRuntime(NamespaceSymbol symbol)
+                {
+                    return symbol.Name == runtime && isSystem(symbol.ContainingNamespace);
+                }
+
+                bool isMicrosoft(NamespaceSymbol symbol)
+                {
+                    return symbol.Name == microsoft && symbol.ContainingNamespace.IsGlobalNamespace;
+                }
+            }
+
+            class InjectedNamespaceSymbol : NamespaceSymbol
+            {
+                private ModuleSymbol _module;
```

Comment:
>private ModuleSymbol _module; [](start = 16, length = 29)

I think this field is not necessary. It looks like it is used only for ```Extent``` and I think we can safely return extent for ```_containingNamespace```